src/message.rs

```diff
@@ -302,6 +302,16 @@ impl MessageKind {
         }
     }
 
+    pub fn get_amount(&self) -> Option<Amount> {
+        if self.action != Action::TakeSell && self.action != Action::TakeBuy {
+            return None;
+        }
+        match &self.content {
```

Comment:
In case this function is oriented to work with TakeSell and TakeBuy actions, you should use `Content::PaymentRequest()` for TakeSell and `Content::Amount()` for TakeBuy

https://docs.rs/mostro-core/latest/mostro_core/message/enum.Content.html

Reply:
Yep will fix!

Reply:
This should be better...

```Rust
pub fn get_amount(&self) -> Option<Amount> {
        if self.action != Action::TakeSell && self.action != Action::TakeBuy {
            return None;
        }
        match &self.content {
            Some(Content::PaymentRequest(_, _, amount)) => *amount,
            Some(Content::Amount(amount)) => Some(*amount),
            _ => None,
        }
    }
```